packages/round-manager/src/features/round/ApplicationsToApproveReject.tsx

```diff
@@ -0,0 +1,323 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { InboxInIcon as NoApplicationsForRoundIcon } from "@heroicons/react/outline";
+import { Spinner } from "../common/Spinner";
+import {
+  BasicCard,
+  CardContent,
+  CardDescription,
+  CardHeader,
+  CardsContainer,
+  CardTitle,
+} from "../common/styles";
+import { Button } from "common/src/styles";
+import {
+  ApplicationStatus,
+  GrantApplication,
+  ProgressStatus,
+  ProgressStep,
+  ProjectStatus,
+} from "../api/types";
+import ConfirmationModal from "../common/ConfirmationModal";
+import {
+  AdditionalGasFeesNote,
+  ApplicationHeader,
+  ApprovedApplicationsCount,
+  Cancel,
+  RejectedApplicationsCount,
+  Select,
+} from "./BulkApplicationCommon";
+import { useApplicationByRoundId } from "../../context/application/ApplicationContext";
+import { datadogLogs } from "@datadog/browser-logs";
+import { useBulkUpdateGrantApplications } from "../../context/application/BulkUpdateGrantApplicationContext";
+import ProgressModal from "../common/ProgressModal";
+import { errorModalDelayMs } from "../../constants";
+import ErrorModal from "../common/ErrorModal";
+import { renderToPlainText } from "common";
+
+type Props = {
+  isDirectRound?: boolean;
+};
+
+export default function ApplicationsToApproveReject({
+  isDirectRound = false,
+}: Props) {
+  const { id } = useParams();
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { applications, isLoading } = useApplicationByRoundId(id!);
```

Comment:
If `id` is not provided, the application will crash without clear reason.

On the other hand, if presence of `id` is validated with something like `if (id === undefined) throw new Error('id not provided')`, application will still crash, but it will be easier to figure out why both for support and for developers.